### PR TITLE
Refactor: Add etcd backend test support and update docs

### PR DESCRIPTION
**Pull request by ahrav**
Moves test support helpers for the etcd coordination backend into a dedicated `backend/test_support.rs` module. Updates documentation for `docs/gossip-coordination/boundary-2-coordination.md` and `docs/gossip-coordination/coordination-testing.md` to reflect the new test support structure and clarify the status of certain operations in the etcd backend. Adjusts the etcd container image used in integration tests to a specific `quay.io/coreos/etcd` version and updates its command arguments for better single-node startup.

## Summary by CodeRabbit

* **New Features**
  * Added gated test-support with fault-injection and snapshot helpers for deterministic etcd coordination tests.

* **Tests**
  * Expanded integration tests covering split atomicity, idempotent replay, abort paths, unpark and collision scenarios; new helpers to seed and inspect state.

* **Chores**
  * Pinned and updated etcd test container startup to a specific upstream image and explicit startup arguments.

* **Documentation**
  * Updated coordination and testing docs to describe new test approach and coverage.

### CODE REVIEW

File: docs/gossip-coordination/boundary-2-coordination.md [hunk 1]
```diff
@@ -32,9 +32,8 @@ used by the in-memory backend. The persisted backend enforces those limits
 before `register_shards`, `split_replace`, and `split_residual` by counting
 existing shard-record keys under the relevant etcd prefixes, then rejecting
 growth that would exceed the configured caps.
-The remaining mutating operations (`complete`, `park_shard`, run terminal
-transitions, `unpark_shard`) fail closed until their persisted transaction
-shapes land.
+The remaining mutating operations (`complete` and `park_shard`) still fail
+closed until their persisted transaction shapes land.
```
[the remark this answers is not in the document]
Comment (ahrav) [reply docs/gossip-coordination/boundary-2-coordination.md]: Good catch — added "shard-level" to disambiguate from run-level mutators that are already persisted.

**Comment by ahrav** [comment]
Re: the outside-diff suggestion to add an integration test driving `split_replace`/`split_residual` at the exact capacity boundary (lines 2797-2870):

The call site at `backend.rs:2046` already passes `sorted.len().saturating_sub(1)` with an inline comment explaining the N-1 adjustment. The unit tests at `tests.rs:2797-2870` prove both the "buggy N" and "correct N-1" arithmetic in isolation. An integration test exercising the real split path at the boundary would add marginal regression protection against someone changing the caller-side value, but the risk is low given the explicit commentary at the call site.

Leaving this for a follow-up if desired — the existing coverage is sufficient for this PR.
